common_examples/tcp_server/main/tcp_server.c

```diff
@@ -0,0 +1,142 @@
+        .lost_ip_event = 0,
+        .flags = ESP_NETIF_DHCP_SERVER,
+        .route_prio = 50
+    };
+    cfg = (esp_netif_config_t) {
+        .base = &eth_netif_cfg,
+        .stack = ESP_NETIF_NETSTACK_DEFAULT_ETH
+    };
+
+    for (uint8_t i = 0; i < eth_port_cnt; i++) {
+        sprintf(if_key_str, "ETH_S%d", i);
+        sprintf(if_desc_str, "eth%d", i);
+
+        esp_netif_ip_info_t ip_info_i = {
+            .ip = {.addr = ESP_IP4TOADDR(192, 168, i, 1)},
+            .netmask = {.addr =  ESP_IP4TOADDR(255, 255, 255, 0)},
+            .gw = {.addr = ESP_IP4TOADDR(192, 168, i, 1)}
+        };
+        ip_infos[i] = ip_info_i;
+
+        eth_netif_cfg.if_key = if_key_str;
+        eth_netif_cfg.if_desc = if_desc_str;
+        eth_netif_cfg.route_prio -= i * 5;
+        eth_netif_cfg.ip_info = &(ip_infos[i]);
+        esp_netif_t *eth_netif = esp_netif_new(&cfg);
+        ESP_ERROR_CHECK(esp_netif_attach(eth_netif, esp_eth_new_netif_glue(eth_handles[i])));
+    }
+    esp_event_handler_register(ETH_EVENT, ETHERNET_EVENT_CONNECTED, start_dhcp_server_after_connection, NULL);
+    ESP_LOGI(TAG, "--------");
+    for (uint8_t i = 0; i < eth_port_cnt; i++) {
+        esp_eth_start(eth_handles[i]);
+        ESP_LOGI(TAG, "Network Interface %d: " IPSTR, i, IP2STR(&ip_infos[i].ip));
+    }
+    ESP_LOGI(TAG, "--------");
+#else
+    if (eth_port_cnt == 1) {
+        // Use default config when using one interface
+        eth_netif_cfg = *(ESP_NETIF_BASE_DEFAULT_ETH);
+    } else {
+        // Set config to support multiple interfaces
+        eth_netif_cfg = (esp_netif_inherent_config_t) ESP_NETIF_INHERENT_DEFAULT_ETH();
+    }
+    cfg = (esp_netif_config_t) {
+        .base = &eth_netif_cfg,
+        .stack = ESP_NETIF_NETSTACK_DEFAULT_ETH
+    };
+    for (int i = 0; i < eth_port_cnt; i++) {
+        sprintf(if_key_str, "ETH_%d", i);
+        sprintf(if_desc_str, "eth%d", i);
+        eth_netif_cfg.if_key = if_key_str;
+        eth_netif_cfg.if_desc = if_desc_str;
+        eth_netif_cfg.route_prio -= i * 5;
+        esp_netif_t *eth_netif = esp_netif_new(&cfg);
+        ESP_ERROR_CHECK(esp_netif_attach(eth_netif, esp_eth_new_netif_glue(eth_handles[i])));
+        esp_eth_start(eth_handles[i]);
+    }
+#endif
+
+    int server_fd, client_fd;
+    struct sockaddr_in server, client;
+    char rxbuffer[SOCKET_MAX_LENGTH] = {0};
+    char txbuffer[SOCKET_MAX_LENGTH] = {0};
+    server_fd = socket(AF_INET, SOCK_STREAM, 0);
+
+    server.sin_family = AF_INET;
+    server.sin_port = htons(SOCKET_PORT);
+    server.sin_addr.s_addr = htonl(INADDR_ANY);
+
+    bind(server_fd, (struct sockaddr *) &server, sizeof(server));
+    listen(server_fd, 1);
+    int transmission_cnt = 0;
+    while (1) {
+        socklen_t client_len = sizeof(client);
+        client_fd = accept(server_fd, (struct sockaddr *) &client, &client_len);
+        if (client_fd == -1) {
+            ESP_LOGE(TAG, "An error has occured when accepting a connection (errno: %d)", errno);
```

Comment:
Nice that you added error handling, but I wonder what happens if indeed we fail to accept socket.
If I read correctly it would pass the control to the infinite loop below and indefinitely log errors, correct?

Reply:
Maybe we should try to accept again
```suggestion
            ESP_LOGE(TAG, "An error has occurred when accepting a connection (errno: %d)", errno);
            continue;
```
or give up
```suggestion
            ESP_LOGE(TAG, "An error has occurred when accepting a connection (errno: %d)", errno);
            break;
```